### PR TITLE
Optimize tutorial

### LINKED ISSUE #957: MVP in-browser inspec demo
Part of this story: https://github.com/chef/inspec/issues/851

Create one step in the in-browser demo. Record against inspec + playback in html for one real steps.

### LINKED ISSUE #1045: Improve InSpec tutorial
### Description

The current tutorial misses some details:
- [x] `clear` (linux) and `cls` (windows) command to clear the terminal
- [x] proper scrolling of terminal
- [x] style of tutorial text looks a little bit off
- [x] sometimes the shell promp is off, i can remove content of the prompt with my back key
- [x] continue to remove keys if i keep holding the back key
- [x] we need to harmonize the recording terminal size = playing terminal size
- [x] copy & paste content

### LINKED ISSUE #1086: www: Online tutorial: safari inspec online demo bug!

### LINKED ISSUE #1089: demo improvements
noticed a few things while testing after https://github.com/chef/inspec/commit/f54fa7bbdfcebb5ee8119cac7c7122a6b36cbe3a merged:  
- [x] inspec shell broken
- [x]  inspec shell -c 'sshd_config.Protocol' returns no output/blank
- [x] we should probably display describe blocks in the instructions exactly as a user would type them
- [x] did we get rid of the control block instructions part?
- [x] maybe fix formatting on last instructions page to make commands there more readable
- [x] --format json | jq command not matching
- [x] handle describe block matching in a better way (leftover task from before)

### PR DESCRIPTION
**Pull request by chris-rock**
This PR is doing some PR improvements:
- copy&paste of commands
- simplify xterm angular 2 abstraction
- move from angular 2 event handling to xterm event handling
- restructure command.json, includes information about the shell
- updates html

Still to do:
- [x] support multiline inspec shell commands
- [x] filter content for inspec shell commands
- [x] update generator and prep docker container for commands like `inspec shell -c 'sshd_config.Protocol'` or `inspec exec examples/inheritance`
- [x] improve html handling
- [x] add all supermarket commands

fixes #1045
fixes #1089
fixes #957
fixes #1086
addresses most issues from #851

screen shot 2016-09-21 at 2 04 10 am

### CODE REVIEW

**Comment by vjeffrey** [comment]
screen shot 2016-09-21 at 10 28 08 am

**Comment by vjeffrey** [comment]
we're no longer clearing the screen on next/prev?

**Comment by arlimus** [comment]
huge work, thank you!!

Let's sync on this, it currently fails to build the static site.

File: www/tutorial/package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "version": "1.0.0",
   "scripts": {
     "build": "node --max-old-space-size=4096 ./node_modules/webpack/bin/webpack --config config/webpack.config.js --progress --profile --colors --display-error-details --display-cached",
-    "start": "node --max-old-space-size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server --config config/webpack.config.js & open http://localhost:8080/",
-    "dev": "node --max-old-space-size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server --config config/webpack.config.js --inline --progress --hot --profile --colors --watch --display-error-details --display-cachedwebpack-dev-server & open http://localhost:8080/"
+    "start": "open http://localhost:8080/ && node --max-old-space-size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server --config config/webpack.config.js",
+    "dev": "open http://localhost:8080/ && node --max-old-space-size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server --config config/webpack.config.js --inline --progress --hot --profile --colors --watch --display-error-details --display-cachedwebpack-dev-server"
```
Comment (arlimus) [inline www/tutorial/package.json]: The `open` command doesn't work across systems. This will essentially fail the build to display anything.

**Comment by arlimus** [comment]
The default seems a bit tiny, there's also a search-bar on the right hand side (not sure this is new, but meh :grin: )

screenshot from 2016-09-21 19-49-34

**Comment by arlimus** [comment]
I really like the small logo, well done!!

**Comment by arlimus** [comment]
Also, I'd prefer not to have any plugins as part of the online demo (but let's discuss).
If we keep them, we should keep these much shorter.

**Comment by arlimus** [comment]
Thank you for the countless improvements in this one!

**Pull request merged**